src/parser/transformer.cpp

```diff
@@ -470,12 +526,16 @@ unique_ptr<ParsedExpression> Transformer::transformPowerOfExpression(
     return expression;
 }
 
-unique_ptr<ParsedExpression> Transformer::transformUnaryAddOrSubtractExpression(
-    CypherParser::OC_UnaryAddOrSubtractExpressionContext& ctx) {
+unique_ptr<ParsedExpression> Transformer::transformUnaryAddSubtractOrFactorialExpression(
+    CypherParser::OC_UnaryAddSubtractOrFactorialExpressionContext& ctx) {
     if (ctx.MINUS()) {
         return make_unique<ParsedFunctionExpression>(NEGATE_FUNC_NAME,
             transformStringListNullOperatorExpression(*ctx.oC_StringListNullOperatorExpression()),
             ctx.getText());
+    } else if (ctx.FACTORIAL()) {
```

Comment:
The input might be `-2!` which contains both MINUS() and FACTORIAL().

Either define their order here (i.e. first create factorial function if necessary then create minus function is necessary) and leave a comment. Or u can define their order in antlr4.

I might do the ordering here because I'm a bit concerned with antlr4 performance as number of grammar rules increasing. But you can make the call here,

Reply:
Done